File: CONSPECTS/easy difficulty/136. Single Number.py
 # Дано непорожній масив цілих чисел nums, де кожен елемент з'являється двічі, за винятком одного. Знайдіть це єдине число.
 # Ви повинні реалізувати розв'язок з лінійною складністю часу та використовувати тільки константний додатковий простір.
 #
 # Example 1:
 # Input: nums = [2,2,1]
 # Output: 1
 #
 # Example 2:
 # Input: nums = [4,1,2,1,2]
 # Output: 4
 #
 # Example 3:
 # Input: nums = [1]
 # Output: 1

nums = [4,1,2,1,2]
dic = {}

for x in nums:
    if x in dic:
        dic[x] += 1
    else:
        dic[x] = 1

for key in dic:
    if dic[key] == 1:
        print(key)

File: CONSPECTS/easy difficulty/136. Single Number.js
 Given a non-empty array of integers nums, every element appears twice except for one. Find that single one.
 You must implement a solution with a linear runtime complexity and use only constant extra space.

 Example 1:
 Input: nums = [2,2,1]
 Output: 1

 Example 2:
 Input: nums = [4,1,2,1,2]
 Output: 4

 Example 3:
 Input: nums = [1]
 Output: 1

var singleNumber = function (nums) {
    const obj = {};
    nums.forEach(num => {
        if (obj.hasOwnProperty(num)) {
            obj[num]++;
        } else {
            obj[num] = 1
        }
    })
    for (prop in obj) {
        if (obj[prop] == 1) return prop
    }
};